blockchain/validators/src/avr_validator.rs

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) 2018-2022 The MobileCoin Foundation
+
+//! Types for reading and writing historical Attestation Verification
+//! Reports from MobileCoin consensus enclaves
+
+use hex::ToHex;
+use mc_attest_core::VerificationReport;
+use mc_common::ResponderId;
+use mc_crypto_keys::Ed25519Public;
+use serde::{Deserialize, Serialize};
+use std::{collections::HashMap, string::String, vec::Vec};
+use toml::{map::Map, value::Value};
+
+/// Object representing AVR history for all consensus nodes
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct AVRRecords {
+    /// AVR Records
+    pub avr_records: HashMap<ResponderId, Vec<AVRRecord>>,
+}
+
+impl AVRRecords {
+    /// Create new records object
+    pub fn new(avr_records: HashMap<ResponderId, Vec<AVRRecord>>) -> Self {
+        Self { avr_records }
+    }
+}
+
+/// Stores a single historical AVR for each tuple of responder_id
+/// & block signing key  in the format specified in draft MCIP #43
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct AVRRecord {
+    /// Uri of the consensus node
+    pub responder_id: ResponderId,
+
+    /// Block signing key of the consensus enclave
+    pub block_signing_key: Option<Ed25519Public>,
+
+    /// Block the AVR for the signing key becomes valid
+    pub first_block_index: u64,
+
+    /// Final block the AVR for the signing key is valid
+    pub last_block_index: u64,
+
+    /// Hex of the AVR bytes
+    pub avr: Option<VerificationReport>,
+}
+
+impl AVRRecord {
+    /// Create new AVRRecord
+    pub fn new(
+        responder_id: &ResponderId,
+        block_signing_key: &Option<Ed25519Public>,
+        first_block_index: u64,
+        last_block_index: u64,
+        avr: Option<VerificationReport>,
+    ) -> Self {
+        Self {
+            responder_id: responder_id.clone(),
+            block_signing_key: *block_signing_key,
+            first_block_index,
+            last_block_index,
+            avr,
+        }
+    }
+}
+
+impl Default for AVRRecords {
+    fn default() -> Self {
+        let avr_records: HashMap<ResponderId, Vec<AVRRecord>> = HashMap::default();
+        Self::new(avr_records)
```

Comment:
How is this different from `derive(Default)`?

If it really is, this is a lot simpler by inlining the var.
```suggestion
        Self::new(Default::default())
```